The "file" argument must be of type string. Received type undefined

I have a directory `dist/`, when I run `gh-pages --dist dist` from my package.json, I get this error:

`The "file" argument must be of type string. Received type undefined`

```
node: v12.7.0
Ubuntu (Pop!_OS 19.04)
```

I tried updating both the package and lock with versions 2.0.1 and 2.1.0 and I am still getting the same error.   I am a bit stumped here.  This is just a basic react app.   Any suggestions...

> gh-pages@2.0.1 deploy C:\Users\Chris\Desktop\URCoding\GitHub_Work\chrisksportfolio
> gh-pages -d build

The "file" argument must be of type string. Received type undefined
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! gh-pages@2.0.1 deploy: `gh-pages -d build`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the gh-pages@2.0.1 deploy script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

--- a/bin/gh-pages.js
+++ b/bin/gh-pages.js
@@ -26,27 +26,35 @@ function main(args) {
       .option(
         '-s, --src <src>',
         'Pattern used to select which files to publish',
-        '**/*'
+        ghpages.defaults.src
       )
       .option(
         '-b, --branch <branch>',
         'Name of the branch you are pushing to',
-        'gh-pages'
+        ghpages.defaults.branch
       )
       .option(
         '-e, --dest <dest>',
         'Target directory within the destination branch (relative to the root)',
-        '.'
+        ghpages.defaults.dest
       )
       .option('-a, --add', 'Only add, and never remove existing files')
       .option('-x, --silent', 'Do not output the repository url')
-      .option('-m, --message <message>', 'commit message', 'Updates')
+      .option(
+        '-m, --message <message>',
+        'commit message',
+        ghpages.defaults.message
+      )
       .option('-g, --tag <tag>', 'add tag to commit')
-      .option('--git <git>', 'Path to git executable')
+      .option('--git <git>', 'Path to git executable', ghpages.defaults.git)
       .option('-t, --dotfiles', 'Include dotfiles')
       .option('-r, --repo <repo>', 'URL of the repository you are pushing to')
-      .option('-p, --depth <depth>', 'depth for clone', 1)
-      .option('-o, --remote <name>', 'The name of the remote', 'origin')
+      .option('-p, --depth <depth>', 'depth for clone', ghpages.defaults.depth)
+      .option(
+        '-o, --remote <name>',
+        'The name of the remote',
+        ghpages.defaults.remote
+      )
       .option(
         '-u, --user <address>',
         'The name and email of the user (defaults to the git config).  Format is "Your Name <email@example.com>".'
@@ -55,7 +63,7 @@ function main(args) {
         '-v, --remove <pattern>',
         'Remove files that match the given pattern ' +
           '(ignored if used together with --add).',
-        '.'
+        ghpages.defaults.only
       )
       .option('-n, --no-push', 'Commit only (with no push)')
       .parse(args);
@@ -81,6 +89,7 @@ function main(args) {
       message: program.message,
       tag: program.tag,
       git: program.git,
+      depth: program.depth,
       dotfiles: !!program.dotfiles,
       add: !!program.add,
       only: program.remove,

--- a/lib/index.js
+++ b/lib/index.js
@@ -22,6 +22,21 @@ function getRepo(options) {
   }
 }
 
+exports.defaults = {
+  dest: '.',
+  add: false,
+  git: 'git',
+  depth: 1,
+  dotfiles: false,
+  branch: 'gh-pages',
+  remote: 'origin',
+  src: '**/*',
+  only: '.',
+  push: true,
+  message: 'Updates',
+  silent: false
+};
+
 /**
  * Push a git branch to a remote (pushes gh-pages by default).
  * @param {string} basePath The base path.
@@ -34,22 +49,7 @@ exports.publish = function publish(basePath, config, callback) {
     config = {};
   }
 
-  const defaults = {
-    dest: '.',
-    add: false,
-    git: 'git',
-    depth: 1,
-    dotfiles: false,
-    branch: 'gh-pages',
-    remote: 'origin',
-    src: '**/*',
-    only: '.',
-    push: true,
-    message: 'Updates',
-    silent: false
-  };
-
-  const options = Object.assign({}, defaults, config);
+  const options = Object.assign({}, exports.defaults, config);
 
   if (!callback) {
     callback = function(err) {

--- a/test/bin/gh-pages.spec.js
+++ b/test/bin/gh-pages.spec.js
@@ -15,24 +15,11 @@ describe('gh-pages', () => {
       ghpages.publish.restore();
     });
 
-    const defaults = {
-      repo: undefined,
-      silent: false,
-      branch: 'gh-pages',
-      src: '**/*',
-      dest: '.',
-      message: 'Updates',
-      dotfiles: false,
-      add: false,
-      remote: 'origin',
-      push: true
-    };
-
     const scenarios = [
       {
         args: ['--dist', 'lib'],
         dist: 'lib',
-        config: defaults
+        config: ghpages.defaults
       },
       {
         args: ['--dist', 'lib', '-n'],

